packages/frontend/src/features/catalog/components/CatalogPanel.tsx

```diff
@@ -414,13 +433,61 @@ export const CatalogPanel: FC = () => {
                 </Group>
             </Group>
 
-            <CatalogTree
-                tree={catalogTree}
-                projectUuid={projectUuid}
-                searchString={debouncedSearch}
-                selection={selection}
-                onItemClick={selectAndGetMetadata}
-            />
+            {noResults ? (
```

Comment:
In other components, we normally return `invalid states` in a different return , instead of adding this conditional state in the main return. What's the best react way to do this ?

Reply:
The thing is that this is below the search controls so we don't want to return a whole different tree, just this part. But this could go in the CatalogTree component and that could return this as an invalid state if you prefer that. I thought about doing it that way, but went with this do sort of centralize the fetch/empty state logic.

Reply:
This is pretty standard react but if there is a pattern about this we like better, doing it differently is fine with me.

Reply:
This is ok, I was wondering if there is a best practice out there for this